### ISSUE TITLE
MD053: false positive for footnotes with multiple lines

### ISSUE THREAD

**Issue by vEnhance** [rvben/rumdl#540]
### rumdl version

0.1.57

### Installation method

cargo install

### Bug description

When a footnote is more than one line, it can trigger false positive [MD053] rule. This doesn't appear to happen if the footnote is only one line long.

### Steps to reproduce

```
$ cat mwe.md
# Greetings

This is a paragraph that has a footnote.[^footnote]

[^footnote]:
    This footnote is long enough that it doesn't fit on just one line.
    Here is my [website][web].

[web]: https://web.evanchen.cc
$ rumdl --version
rumdl 0.1.57
$ rumdl check --no-config mwe.md
mwe.md:9:1: [MD053] Unused link/image reference: [web]

Issues: Found 1 issues in 1 file (4ms)
```

### Expected behavior

MD053 should not appear since it is processed as part of a footnote.

### Actual behavior

MD053 was flagged.

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.1.59 (just released).

The root cause was a parser option mismatch: the code block detector used `ENABLE_OLD_FOOTNOTES` which treated 4-space-indented footnote continuation lines as indented code blocks, hiding them from MD053's reference usage scan.

The fix switches to `ENABLE_FOOTNOTES`, adds proper footnote body detection (`in_footnote_definition`), and skips footnote content in MD005, MD007, MD030, MD032, and MD046 to prevent related false positives.

Thanks for the clear reproduction case, @vEnhance.
